libraries/AC_WPNav/AC_Loiter.cpp

```diff
@@ -309,12 +282,19 @@ void AC_Loiter::calc_desired_velocity(float nav_dt, bool avoidance_on)
         AC_Avoid *_avoid = AP::ac_avoid();
         if (_avoid != nullptr) {
             Vector3f avoidance_vel_3d{desired_vel.x, desired_vel.y, 0.0f};
-            _avoid->adjust_velocity(avoidance_vel_3d, _pos_control.get_pos_xy_p().kP(), _accel_cmss, _pos_control.get_pos_z_p().kP(), _pos_control.get_max_accel_z(), nav_dt);
+            _avoid->adjust_velocity(avoidance_vel_3d, _pos_control.get_pos_xy_p().kP(), _accel_cmss, _pos_control.get_pos_z_p().kP(), _pos_control.get_max_accel_z_cmss(), nav_dt);
             desired_vel = Vector2f{avoidance_vel_3d.x, avoidance_vel_3d.y};
         }
     }
 
+    // get loiters desired velocity from the position controller where it is being stored.
+    const Vector3f &target_pos_3d = _pos_control.get_pos_target_cm();
+    Vector2f target_pos(target_pos_3d.x,target_pos_3d.y);
```

Comment:
I'm sure PeterB would prefer we use curly braces for this line.  So make it, "Vector2f target_pos{target_pos_3d.x,target_pos_3d.y};".